default/AI/ResourcesAI.py

```diff
@@ -29,98 +38,261 @@
+                oldFocus = pinfo.current_focus
+                newFocus = pinfo.future_focus
+                cPP, cRP = pinfo.current_output[IFocus], pinfo.current_output[RFocus]
+                otPP, otRP = pinfo.possible_output.get(oldFocus, (0, 0))
+                ntPP, ntRP = pinfo.possible_output[newFocus]
+                print (Reporter.table_format %
+                       ("pID (%3d) %22s" % (pid, pinfo.planet.name[-22:]),
+                        "c: %5.1f / %5.1f" % (cRP, cPP),
+                        "cT: %5.1f / %5.1f" % (otRP, otPP),
+                        "cF: %8s" % fociMap.get(oldFocus, 'unknown'),
+                        "nF: %8s" % fociMap.get(newFocus, 'unset'),
+                        "cT: %5.1f / %5.1f" % (ntRP, ntPP)))
+        self.print_table_footer(priorityRatio)
+
+
+    @staticmethod
+    def print_resource_ai_footer():
+        empire = fo.getEmpire()
+        aPP, aRP = empire.productionPoints, empire.resourceProduction(fo.resourceType.research)
+        # Next string used in charts. Don't modify it!
+        print "Current Output (turn %4d) RP/PP : %.2f ( %.1f / %.1f )" % (fo.currentTurn(), aRP / (aPP + 0.0001), aRP, aPP)
+        print "------------------------"
+        print "ResourcesAI Time Requirements:"
+
+
+    @staticmethod
+    def print_resources_priority():
+        """calculate top resource priority"""
+        universe = fo.getUniverse()
+        empire = fo.getEmpire()
+        empirePlanetIDs = PlanetUtilsAI.get_owned_planets_by_empire(universe.planetIDs)
+        print "Resource Management:"
         print
-    warnings.clear()
+        print "Resource Priorities:"
+        resourcePriorities = {}
+        for priorityType in get_priority_resource_types():
+            resourcePriorities[priorityType] = foAI.foAIstate.get_priority(priorityType)
+
+        sortedPriorities = resourcePriorities.items()
+        sortedPriorities.sort(lambda x, y: cmp(x[1], y[1]), reverse=True)
+        topPriority = -1
+        for evaluationPair in sortedPriorities:
+            if topPriority < 0:
+                topPriority = evaluationPair[0]
+            print "    ResourcePriority |Score: %s | %s " % (evaluationPair[0], evaluationPair[1])
+
+        # what is the focus of available resource centers?
+        print
+        warnings = {}
+        #TODO combine this with previous table to reduce report duplication?
+        print "Planet Resources Foci:"
+        for planetID in empirePlanetIDs:
+            planet = universe.getPlanet(planetID)
+            planetPopulation = planet.currentMeterValue(fo.meterType.population)
+            maxPop = planet.currentMeterValue(fo.meterType.targetPopulation)
+            if maxPop < 1 and planetPopulation > 0:
+                warnings[planet.name] = (planetPopulation, maxPop)
+            statusStr = "  ID: " + str(planetID) + " Name: % 18s -- % 6s % 8s " % (str(planet.name), str(planet.size), str(planet.type))
```

Comment:
Mix of formatting and string concatenation in single line?  

`str(planet.name), str(planet.size), str(planet.type)` you use `%s`, no need to wrap arguments to string.